File: LeetCode/Easy/0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] ans = new int[n+1];

        for (int i = 0; i <= n; i++) {
            ans[i] = numberOfSetBit(i);
        }

        return ans;
    }

    public int numberOfSetBit(int n) {
        int cnt = 0;
        while (n != 0) {
            if (n % 2 == 1) cnt ++;
            n = (int) n / 2;
        }
        return cnt;
    }
}

File: LeetCode/Easy/0338-counting-bits/0338-counting-bits.go
func countBits(n int) []int {
    ans := make([]int, 0)
    for i := 0; i < n + 1; i ++ {
        ans = append(ans, countSetBit(i))
    }
    return ans
}

func countSetBit(x int) int {
    cnt := 0
    for (x != 0) {
        if (x % 2 == 1) {
            cnt += 1
        }
        x = x/2
    }
    return cnt
}